next/components/forms/segments/NavBar/NavBar.tsx

```diff
@@ -78,6 +78,12 @@ export const NavBar = ({
   const { getRouteWithCurrentUrlRedirect } = useQueryParamRedirect()
   const { userAttributes, isSignedIn, isLegalEntity } = useSsrAuth()
 
+  // Hide taxes and fees section for legal entities
+  if (isLegalEntity) {
+    // eslint-disable-next-line no-param-reassign
+    sectionsList = sectionsList?.filter((section) => section.url !== ROUTES.TAXES_AND_FEES)
```

Comment:
Please don't modify props coming to a component. Is there reason why we cannot filter it out where the list is originally coming from (probably a different component)?

Reply:
My mistake, moved the filtering to parent component